app/api/routes.py

```diff
@@ -32,18 +32,24 @@ def get_plant(id):
 def add_to_garden():
     plant_id = request.args['plant_id']
     plant_name = request.args['plant_name']
-
     plant = Plant.query.filter_by(id=plant_id,).first()
-    days_to_harvest = (datetime.now() + timedelta(days=plant.harvest_time))
 
-    garden = Garden(plant_id=plant_id,plant_name=plant_name)
+    if plant.harvest_time:
+        harvest_date = (datetime.now() + timedelta(days=plant.harvest_time))
+        days_to_harvest = harvest_date.strftime("%a, %B %d %Y")
+    else:
+        days_to_harvest = None
```

Comment:
This was quite the puzzle at first, glad we happened to test on Cactus Dan, id:10 who would fail.  Turns out this original datetime calculation would break the api if a null value was returned.  It was calling datetime functions on null and got all angry.